src/plugins/kibana/server/routes/api/ingest/register_simulate.js

```diff
@@ -18,13 +19,16 @@ export function registerSimulate(server) {
       const simulateApiDocument = request.payload;
       const body = ingestSimulateApiKibanaToEsConverter(simulateApiDocument);
 
+      const handleResponse = processESIngestSimulateResponse;
+      const handleError = _.partial(processESIngestSimulateError, simulateApiDocument.dirty_processor_id);
+
       return boundCallWithRequest('transport.request', {
         path: '_ingest/pipeline/_simulate',
         query: { verbose: true },
         method: 'POST',
         body: body
       })
-      .then(_.partial(processESIngestSimulateResponse, _.map(simulateApiDocument.processors, keysToCamelCaseShallow)))
+      .then(handleResponse, handleError)
```

Comment:
This assumes that the only error the request might throw is a pipeline compile error specific to an individual processor. What if the compile error isn't specific to one processor (e.g. the json synatx is malformed) or if the error is completely unrelated to pipelines (e.g. elasticsearch is unreachable)?

I would recommend using the `handle_er_error.js` module as a fallback for handling the generic errors.